contentcuration/contentcuration/management/commands/restore_channel.py

```diff
@@ -0,0 +1,216 @@
+          sys.stderr.write("\n\n********** STARTING CHANNEL RESTORATION **********\n")
+          start = datetime.datetime.now()
+          channel_id = options['channel_id']
+          target_id = options.get('target_id') or channel_id
+
+          # Test connection to database
+          sys.stderr.write("\nConnecting to database for channel {}...".format(channel_id))
+          conn = sqlite3.connect(os.path.join(settings.DB_ROOT,'{}.sqlite3'.format(channel_id)))
+          cursor=conn.cursor()
+          sys.stderr.write(" DONE")
+
+
+          # Start by creating channel
+          sys.stderr.write("\nCreating channel...")
+          channel, root_pk = create_channel(conn, target_id)
+
+          # Create nodes mapping to channel
+          sys.stderr.write("\n   Creating nodes...")
+          root = None
+          with transaction.atomic():
+            root = create_nodes(cursor, target_id)
+
+          # Save tree to target tree
+          channel.main_tree = root
+          channel.save()
+
+          # Print stats
+          sys.stderr.write("\n\nChannel has been restored (time: {ms}, {node_count} nodes, {file_count} files, {tag_count} tags)".format(
+            ms = datetime.datetime.now() - start,
+            node_count = NODE_COUNT,
+            file_count = FILE_COUNT,
+            tag_count = TAG_COUNT)
+          )
+          sys.stderr.write("\n\n********** RESTORATION COMPLETE **********\n\n")
+
+        except EarlyExit as e:
+            logging.warning("Exited early due to {message}.".format(
+                message=e.message))
+            self.stdout.write("You can find your database in {path}".format(
+                path=e.db_path))
+
+
+def create_channel(cursor, target_id):
+    id, name, description, author, thumbnail, root_pk, version = cursor.execute('SELECT * FROM {table}'.format(table=CHANNEL_TABLE)).fetchone()
+    channel, is_new = models.Channel.objects.get_or_create(pk=target_id)
+    channel.name = name
+    channel.description = description
+    channel.thumbnail = write_to_thumbnail_file(thumbnail)
+    channel.version = version
+    channel.save()
+    sys.stderr.write("\n\tCreated channel {} with name {}".format(target_id, name))
+    return channel, root_pk
+
+def write_to_thumbnail_file(raw_thumbnail):
+    if raw_thumbnail and raw_thumbnail != "":
+        with SimpleUploadedFile('temp.png',raw_thumbnail.replace('data:image/png;base64,', '').decode('base64')) as tempf:
+            filename = write_file_to_storage(tempf, check_valid=False)
+            sys.stderr.write("\n\tCreated thumbnail {}".format(filename))
+            return filename
+
+def create_nodes(cursor, target_id, parent=None, indent=1):
+
+    # Read database rows that match parent
+    parent_query = 'parent_id IS NULL'
+    if parent:
+        parent_query = "parent_id=\'{}\'".format(parent.node_id)
+
+    sql_command = 'SELECT id, title, content_id, description, sort_order, '\
```

Comment:
I'm not comfortable with raw SQL calls, but will defer to you if you think this is the best way on efficiency vs. idiomaticness.